packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -485,31 +510,39 @@ class _Label extends StatelessWidget {
     @required this.colorTween,
     @required this.animation,
     @required this.item,
-    @required this.selectedFontSize,
-    @required this.unselectedFontSize,
+    @required this.selectedLabelStyle,
+    @required this.unselectedLabelStyle,
     @required this.showSelectedLabels,
     @required this.showUnselectedLabels,
   }) : assert(colorTween != null),
        assert(animation != null),
        assert(item != null),
-       assert(selectedFontSize != null),
-       assert(unselectedFontSize != null),
+       assert(selectedLabelStyle != null),
+       assert(unselectedLabelStyle != null),
        assert(showSelectedLabels != null),
        assert(showUnselectedLabels != null),
        super(key: key);
 
   final ColorTween colorTween;
   final Animation<double> animation;
   final BottomNavigationBarItem item;
-  final double selectedFontSize;
-  final double unselectedFontSize;
+  final TextStyle selectedLabelStyle;
+  final TextStyle unselectedLabelStyle;
   final bool showSelectedLabels;
   final bool showUnselectedLabels;
 
   @override
   Widget build(BuildContext context) {
+    final double selectedFontSize = selectedLabelStyle.fontSize;
+    final double unselectedFontSize = unselectedLabelStyle.fontSize;
+
+    final TextStyle customStyle = TextStyle.lerp(
+      unselectedLabelStyle,
+      selectedLabelStyle,
+      animation.value,
+    );
     Widget text = DefaultTextStyle.merge(
```

Comment:
This really is a cool example of mixing OOP and reactive programming. A case where we have to process an instance a lot before returning it.

Reply:
Thanks, yeah calling `lerp` manually is pretty interesting.